### ISSUE TITLE
NREPEAT has different meaning in recipe and pipeline.

### ISSUE THREAD

**Issue by bberkeyU** [NCAR/ucomp-pipeline#92]
The header value NREPEAT is insensitive to which camera is onband.  In the recipe definition, we consider the number of repeats to be the number of times we repeat the exact same Lyot filter config.

Such that the following commands would not be considered a repeat since there is a camera swap between the two commands.
DATA	TCAM	BOTH	1074.70	   16
DATA	RCAM	BOTH	1074.70	   16

It looks like the pipeline considers these to be the same command because in the pipeline NREPEATS = # unique wavelengths/ #extensions for a given fits file.

I am not sure if there is a right answer for this as both definitions make sense in their context.   Maybe we just need to document the differences.

**Comment by mgalloy** [comment]
This looks like a difference between my and Steve's pipelines. For my pipeline:

    level1$ ucomp cat -r 20221101.173926.ucomp.530.l1.3.fts | grep -i NREPEAT
    NREPEAT =                   16 / number of repeats of wavelength scans

For Steve's:

    level1$ ucomp cat -r 20221101.173926.ucomp.530.l1.3.fts | grep -i NREPEAT
    NREPEAT =                    8 / Number of repeats of wavelength scans

I think I just need to divide by 2, i.e., the number of onboard states.

As far as I can tell, the raw data doesn't have `NREPEAT` in the headers. Is that correct, @bberkeyU?

**Comment by mgalloy** [comment]
Maybe I should check the data for these assumptions when doing my inventory? I am not sure what would happen in the pipeline now if there was only one camera, or only one `ONBAND` value, in the level 0 data, but I doubt it would handle it gracefully. I'm pretty sure that it would crash if the raw extensions weren't of size 1280 by 1024 by 4 by 2. I am not sure what would happen if there were only a single `ONBAND` value.

**Comment by bberkeyU** [comment]
Good point, I don't think the pipeline can properly process data that doesn't flip/flip between onbeam and a continuum.    

It is probally fine to divide by 2 here and let the inventory sort out edge cases that are missing either onbeam or continuum.

**Comment by mgalloy** [comment]
I am not sure of all the possible/likely scenarios:

1. It's possible to take data from only a single camera, but alternate `ONBAND`, right? I believe if you set `cameras/use` to "RCAM" or "TCAM" in the configuration file, that is what happens now. The pipeline wouldn't figure it out on its own if `cameras/use` was "BOTH", but only one camera's data was present. The pipeline could be modified to figure this out, though, if that is a likely and/or valuable scenario.
2. Take onband data only from a single camera. Then no continuum subtraction should be done. The pipeline would definitely not handle this. Not sure how valuable data that haven't been continuum subtracted would be.

Yes, it would need to check that there are an equal number of `RCAM` and `TCAM` images per wavelength.

**Comment by bberkeyU** [comment]
I think handling case 1 could be a possibility in the future, but I think case 2 should always be rejected by the pipeline at inventory.  Other than 1083 (which I think we have now rejected), all of our assumptions about what L1 and L2 mean assume a continuum subtracted, which we don't get in case 2.
I guess we could imagine case 3 in which we always get the onband from tcam and continuum from rcam.  But I can't see this ever being something that we would want to run.

In Case 1 or case 0 (our normal operations) if a file passes the pipeline inventory step dividing by 2 will always be the ring thing to do.


, but I don't think data taken in scenario 2 can/should be processed by the science pipeline.
